tests/integ/scala/test_dataframe_writer_suite.py

```diff
@@ -139,6 +135,56 @@ def test_negative_write_with_target_column_name_order(session):
         session.table(table_name).drop_table()
 
 
+def test_write_with_target_column_name_order_all_kinds_of_dataframes_without_truncates(
+    session,
+):
+    table_name = Utils.random_table_name()
+
+    def create_table():
+        session.create_dataframe(
+            [],
+            schema=StructType(
+                [StructField("a", IntegerType()), StructField("b", IntegerType())]
+            ),
+        ).write.save_as_table(table_name, table_type="temporary")
+
+    create_table()
+    try:
+        df1 = session.create_dataframe([[1, 2]], schema=["b", "a"])
+        # DataFrame.cache_result()
+        df_cached = df1.cache_result()
+        df_cached.write.save_as_table(
+            table_name, mode="append", column_order="name", table_type="temp"
+        )
+        Utils.check_answer(session.table(table_name), [Row(**{"A": 2, "B": 1})])
+    finally:
+        session.table(table_name).drop_table()
+
+    create_table()
+    try:
+        # copy DataFrame
+        df_cloned = copy.copy(df1)
```

Comment:
Question: why do we need this test case?

Reply:
I actually was unsure what the same test case with a cloned df. Was part of the code that I reused from the next test which is being skipped. Can totally remove.